### PR TITLE
Fix fuzzy notification icon

### LINKED ISSUE #1481: Notification number pop-up icon edges appear fuzzy
## Bug Report

**Current Behavior**
The notification number pop-up icon looks fuzzy compared to other icons.
This problem existed with other icons before too (#569)

**Steps to Reproduce**
1. Post something
2. Wait for a reply
3. See notification count badge

**Expected Behavior**
The icon corners shouldn't be fuzzy.

**Screenshots**
screenshot_2018-07-03 1 flarum community

**Environment**
Flarum Community
https://discuss.flarum.org

**Possible Solution**
https://github.com/flarum/core/commit/dc757fae5f4a43630447b4c6efc5fe4fbec02a53

This was used for the previous icon bug.

### PR DESCRIPTION
**Pull request by TurboProgramming**
**Fixes #1481**

**Reviewers should focus on:**
Line 40

**Screenshot**
See #1481

### CODE REVIEW

**Comment by datitisev** [comment]
I have two things to say:
1) The `border-radius` should not be removed
2) The result isn't great

image

My solution for this would be `box-shadow: 0px 0px 0px 1px @header-bg` + 1px more padding at the top to mimic the border effect. The final padding for the top would be `3px` (right now it's 2px), resulting in the following:
image

**Comment by TurboProgramming** [comment]
```
.NotificationsDropdown-unread {
 position:absolute;
 top:1px;
 left:17px;
 background:#667199;
 color:#fff;
 font-size:11px;
 font-weight:bold;
 padding:2px 4px 3px;
 line-height:1em;
 border-radius:10px;
 border:1px solid #fff;
 min-width:18px;
 height:18px;
 text-align:center
 box-shadow: 0px 0px 0px 1px @header-bg
}
```

The code above result in this:

screenshot_2018-07-06 flightsimtr forum

**Comment by TurboProgramming** [comment]
```.NotificationsDropdown-unread { position:absolute; top:1px; left:17px; background:#667199; color:#fff; font-size:11px; font-weight:bold; padding:2px 4px 3px; line-height:1em; border-radius:10px; border:1px solid #fff; min-width:18px; height:18px; text-align:center box-shadow: 0px 0px 0px 1px @header-bg}```

**Comment by TurboProgramming** [comment]
I'm also aware that border-radius should stay, I removed it by accident.

**Comment by TurboProgramming** [comment]
When I zoom the page to 240%, it looks like this:

screenshot_2018-07-06 flightsimtr forum 1

Still some errors, but better than the old one.

But when its 100%, the same result, doesn't fix anything.

**Comment by datitisev** [comment]
You didn't remove the border in your example. Make sure to set `padding-top: 3px` as well.

**Comment by datitisev** [comment]
The indentation is now messed up, and the less is not valid.
There should not be any closing bracket.

image

**Comment by TurboProgramming** [comment]
@datitisev Yep, I know.

Check again.

**Comment by datitisev** [comment]
Still messed up, the border is still there, and there should not be any closing bracket.
The indentation should be 2 spaces, not one.
There's no need for extra line breaks either.

image

**Comment by TurboProgramming** [comment]
@datitisev What about now?

**Comment by datitisev** [comment]
Indentation is still wrong. And make sure you only change the line that previously had the `border` so there's not a useless diff.

image

**Comment by franzliedke** [comment]
@TurboProgramming Please take a look at [the "Changes" tab](https://github.com/flarum/core/pull/1500/files) and then keep pushing until the only changes are removing the border, adding the shadow and the padding (as @datitisev suggested).

**Comment by TurboProgramming** [comment]
@datitisev @franzliedke Can you guys check it now? I think it's right now.

**Comment by datitisev** [comment]
Sorry, I just realized that `padding: 2px 4px 3px;` and `padding-top: 3px;` can actually be merged into `padding: 3px 4px;`.
Otherwise LGTM! While you are at it, remove those two spaces on line 45/46 (the empty one) 🙂

**Comment by TurboProgramming** [comment]
@datitisev Completed. 

Can someone preview this live and send a screenshot so we can confirm that this fixes the bug before merging?

**Comment by TurboProgramming** [comment]
> Can someone preview this live and send a screenshot so we can confirm that this fixes the bug before merging?

OK, so I tested it. Here is the result:

screenshot_2018-07-07 flightsimtr - havacilik forumu

**Comment by datitisev** [comment]
Hm... something must be wrong in your environment, as we aren't changing neither the background or the text color.
The preview was what I sent earlier.

**Comment by TurboProgramming** [comment]
So, is this ready to be merged?

File: less/forum/NotificationsDropdown.less [hunk 1]
```diff
@@ -35,14 +35,13 @@
   color: @header-bg;
   font-size: 11px;
   font-weight: bold;
-  padding: 2px 4px 3px;
+  padding: 3px 4px;
   line-height: 1em;
   border-radius: 10px;
-  border: 1px solid @header-bg;
+  box-shadow: 0px 0px 0px 1px @header-bg;
```
Comment (franzliedke) [inline less/forum/NotificationsDropdown.less]: This should probably use the `.box-shadow` mixin just like the other commit did.

Comment (TurboProgramming) [reply less/forum/NotificationsDropdown.less]: I made the change, can you check whether it is OK?

Comment (luceos) [reply less/forum/NotificationsDropdown.less]: @TurboProgramming have you tried your edit on a local installation by any chance? I could be wrong, but from my experience mixins are most commonly included as (a kind of) function calls, see http://lesscss.org/#mixins

Comment (TurboProgramming) [reply less/forum/NotificationsDropdown.less]: @luceos Unfortunately, no. And I don't think I can at the moment. I'll try my best, but if possible can anyone of you do it?

File: less/forum/NotificationsDropdown.less [hunk 2]
```diff
@@ -35,14 +35,13 @@
   color: @header-bg;
   font-size: 11px;
   font-weight: bold;
-  padding: 2px 4px 3px;
+  padding: 3px 4px;
   line-height: 1em;
   border-radius: 10px;
-  border: 1px solid @header-bg;
+  .box-shadow: 0px 0px 0px 1px @header-bg;
```
Comment (tobscure) [inline less/forum/NotificationsDropdown.less]: Still not quite right. Syntax should be:

```less
.box-shadow(0px 0px 0px 1px @header-bg);
```

File: less/forum/NotificationsDropdown.less [hunk 3]
```diff
@@ -35,14 +35,13 @@
...
   min-width: 18px;
   height: 18px;
   text-align: center;
-
```
Comment (tobscure) [inline less/forum/NotificationsDropdown.less]: please don't remove this whitespace

**Comment by tobscure** [comment]
Fixed in https://github.com/flarum/core/commit/fb9a89f67ffc3cc81ed765d459745ca8dc73c229, required a few more adjustments to maintain the exact same appearance as when using the border.

**Pull request closed without merging**